mParticle-Apple-SDK/Data Model/MPForwardRecord.m

```diff
@@ -112,19 +112,22 @@ - (instancetype)initWithMessageType:(MPMessageType)messageType execStatus:(MPKit
     if (kitFilter.appliedProjections.count > 0) {
         NSMutableArray *projections = [[NSMutableArray alloc] initWithCapacity:kitFilter.appliedProjections.count];
         NSMutableDictionary *projectionDictionary;
+        NSString *currentProjectionName = ((MPEvent *)kitFilter.originalEvent).name;
```

Comment:
You should check that `originalEvent` is an `MPEvent` before casting it.

Reply:
added as well